pkgs/by-name/co/conan_1/package.nix

```
@@ -0,0 +1,198 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchPypi
+, git
+, pkg-config
+, python3
+, zlib
+}:
+
+let
+  python = python3.override {
+    packageOverrides = self: super: {
+      # conan v1.xx requires old semver instead of node-semver
+      semver = super.semver.overridePythonAttrs (old: rec {
+        version = "0.6.1";
+        src = fetchPypi {
+          inherit (old) pname;
+          inherit version;
+          sha256 = "4016f7c1071b0493f18db69ea02d3763e98a633606d7c7beca811e53b5ac66b7";
+        };
+        doCheck = false;
+      });
+    };
+  };
+in
+python.pkgs.buildPythonApplication rec {
+  pname = "conan";
+  version = "1.63.0";
+  format = "setuptools";
```

```suggestion
  pyproject = true;
```
and add `setuptools` to `nativeBuildInputs`

---

This is complicated or I'm doing something wrong.

If I just add `pyproject = true;`, I'm getting the error:
<details>
<summary> urllib3<1.27,>=1.26.6 not satisfied by version 2.1.0 </summary> 

```
Checking runtime dependencies for conan-1.63.0-py3-none-any.whl
  - urllib3<1.27,>=1.26.6 not satisfied by version 2.1.0
error: builder for '/nix/store/7sq7paxrngi9fac6gkiinp4vbx2i5gs1-conan-1.63.0.drv' failed with exit code 1;
       last 10 log lines:
       > Finished creating a wheel...
       > /private/tmp/nix-build-conan-1.63.0.drv-2/source/dist /private/tmp/nix-build-conan-1.63.0.drv-2/source
       > Unpacking to: unpacked/conan-1.63.0...OK
       > Repacking wheel as ./conan-1.63.0-py3-none-any.whl...OK
       > /private/tmp/nix-build-conan-1.63.0.drv-2/source
       > Finished executing pypaBuildPhase
       > Running phase: pythonRuntimeDepsCheckHook
       > Executing pythonRuntimeDepsCheck
       > Checking runtime dependencies for conan-1.63.0-py3-none-any.whl
       >   - urllib3<1.27,>=1.26.6 not satisfied by version 2.1.0
```

</details>

Then to fix this I'm adding 

<details>
<summary> override for urllib3 </summary>

```nix
      urllib3 = super.urllib3.overridePythonAttrs (old: rec {
        version = "1.26.6";
        doCheck = false;
      });
```

</details>

And it causes building from source and testing for many python packages which doesn't look right.

Could you please advise how to fix this?

---

You want to drop the strict version requirement on requests instead.

Also this is not related to the changed line but a treewide python change that was done in the meantime.